bridge-sdk/src/nep141/mod.rs

```diff
@@ -0,0 +1,450 @@
+use std::{str::FromStr, sync::Arc};
+use borsh::BorshSerialize;
+use ethers::{abi::Address, prelude::*};
+use near_crypto::SecretKey;
+use near_primitives::{hash::CryptoHash, types::{AccountId, TransactionOrReceiptId}};
+use crate::{common::{Env, Error, Result}, eth_proof_generator, near_on_eth_client::NearOnEthClient, near_rpc_client};
+use light_client_proof::LightClientExecutionProof;
+
+mod light_client_proof;
+
+abigen!(
+    BridgeTokenFactory,
+    r#"[
+      function newBridgeToken(bytes memory proofData, uint64 proofBlockHeight) external returns (address)
+      function deposit(bytes memory proofData, uint64 proofBlockHeight) external
+      function withdraw(string memory token, uint256 amount, string memory recipient) external
+      function nearToEthToken(string calldata nearTokenId) external view returns (address)
+    ]"#
+);
+
+abigen!(
+    ERC20,
+    r#"[
+      function allowance(address _owner, address _spender) public view returns (uint256 remaining)
+      function approve(address spender, uint256 amount) external returns (bool)
+    ]"#
+);
+
+pub struct Nep141Bridging {
+    eth_endpoint: Option<String>,
+    eth_chain_id: u64,
+    eth_private_key: Option<String>,
+    bridge_token_factory_address: String,
+    near_endpoint: Option<String>,
+    near_private_key: Option<String>,
+    near_signer: Option<String>,
+    token_locker_id: String,
+    environment: Env,
+}
+
+impl Nep141Bridging {
+    pub fn new(env: Env) -> Self {
+        match env {
+            Env::Testnet => Self {
+                eth_chain_id: 11155111,
+                bridge_token_factory_address: "0xa9108f7F83Fb661e611991116D526fCa1a9585ab".to_string(),
+                eth_endpoint: None,
+                eth_private_key: None,
+                near_endpoint: None,
+                near_private_key: None,
+                near_signer: None,
+                token_locker_id: "ft-locker.sepolia.testnet".to_string(),
+                environment: env,
```

Comment:
I've decided to include these static values into the binary because we may want to allow users to integrate with an SDK without any additional configuration files. I did want to hear others` opinion on this

Reply:
Fixed